sdk/typespec/typespec_macros/README.md

```diff
@@ -0,0 +1,81 @@
+# TypeSpec Macros crate for Rust
+
+The TypeSpec Macros crate provides procedural macros for [TypeSpec](https://typespec.io)-generated client libraries. These macros simplify the implementation of common patterns required when working with TypeSpec-generated code.
+
+[Source code] | [Package (crates.io)] | [API reference documentation] | [TypeSpec documentation]
+
+## Getting started
+
+> **Note:** This crate is primarily intended for internal use by the `typespec_client_core` crate. Direct usage is discouraged unless you are contributing to or debugging the TypeSpec ecosystem.
+### Install the package
+
+> **Disclaimer:** The following instructions are intended for advanced users or contributors who need to work directly with the `typespec_macros` crate. Most users should depend on the `typespec_client_core` crate instead.
+
+```bash
+cargo add typespec_macros
+```
+
+## Key concepts
+
+This crate provides the following derive macros:
+
+- `Model`: A derive macro that implements the `Model` trait, allowing a type to be deserialized from an HTTP response body.
+- `SafeDebug`: A derive macro that implements debug formatting in a way that avoids leaking personally identifiable information (PII).
+
+### The Model derive macro
+
+The `Model` derive macro is used to implement the `Model` trait for structs that represent data returned from an API. It handles the deserialization of HTTP response bodies into your model types.
+
+### The SafeDebug derive macro
+
+The `SafeDebug` derive macro creates a `Debug` implementation that respects the `#[sensitive]` attribute on struct fields. Fields marked with this attribute will not have their values printed in debug output, protecting potentially sensitive information.
+
+## Examples
+
+### Using the Model derive macro
+
+```rust
+use typespec_macros::Model;
```

Comment:
This and the `SafeDebug` trait should be imported from `typespec_client_core`, not form this crate. The actual traits aren't even defined in the `typespec_macro` so this wouldn't even compile.